create_es_bulk_json.py

```diff
@@ -1,30 +1,43 @@
 # coding=utf-8
-import sys, json, subprocess
+import argparse
+import json
+import subprocess
+import sys
+
 
 # SSH into a remote server and find war files under certain directories
-def get_file_names(instance):
-    path = "*/%s/*/current/dist/*.war" % instance
-    ssh = subprocess.Popen(['ssh', '-l', xe_user, xe_host, 'find', banner_home, '-type l -wholename', path, '-exec readlink {} \;'],
-                           shell=False,
-                           stdout=subprocess.PIPE,
-                           stderr=subprocess.PIPE)
+# Returns a list of absolute paths and/or file names depending on the location
+# of files
+def get_file_paths(instance):
+    path = f"*/{instance}/*/current/dist/*.war"
+    command = [
+        'ssh', '-l', xe_user, xe_host, 'find', banner_home,
+        '-type l -wholename', path, '-exec readlink {} \;'
+    ]
+    ssh = subprocess.Popen(
+        command, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+        encoding='utf8'
+    )
     result = ssh.stdout.readlines()
     # If we didn't get any files, something's wrong
-    if result == []:
+    if not result:
         error = ssh.stderr.readlines()
-        print >> sys.stderr, "ERROR: %s" % error
-        sys.exit(1)
+        sys.exit(f"ERROR: {error}")
+
+    return result
 
-    return result    
 
 # Parse the whole filename into the application name and version.
-def parse_file_names(instance, xe_apps):
-    file_names = get_file_names(instance)
+def parse_file_names(instance):
+    file_paths = get_file_paths(instance)
+
+    # Strip absolute path so only the file name remains
+    file_names = list(map(lambda x: x.split("/")[-1], file_paths))
 
     for file in file_names:
-        app = file.split('-', 1)[0]
-        version = file.split('-', 1)[-1]
-        version = version.split('.war', 1)[0]
+        name_parts = file.split("-")
+        app = name_parts[0]
+        version = name_parts[1].split(".war")[0]
```

Comment:
My assumptions here are that the app name will always be the first substring of the filename ended with a "-" and the version will always be the second substring either ended with a "-" or ".war". Are these reasonable or are there cases that these won't hold up?

Reply:
Those are good assumptions.